watertap/costing/units/osmotically_assisted_reverse_osmosis.py

```diff
@@ -0,0 +1,82 @@
+###############################################################################
+# WaterTAP Copyright (c) 2021, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National
+# Laboratory, National Renewable Energy Laboratory, and National Energy
+# Technology Laboratory (subject to receipt of any required approvals from
+# the U.S. Dept. of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#
+###############################################################################
+
+import pyomo.environ as pyo
+from idaes.core.util.exceptions import ConfigurationError
+from idaes.core.util.misc import StrEnum
+from ..util import cost_membrane, register_costing_parameter_block
+
+
+class OAROType(StrEnum):
+    standard = "standard"
+    high_pressure = "high_pressure"
+
+
+def build_osmotically_assisted_reverse_osmosis_cost_param_block(blk):
+
+    blk.factor_membrane_replacement = pyo.Var(
+        initialize=0.15,
+        doc="Membrane replacement factor [fraction of membrane replaced/year]",
+        units=pyo.units.year**-1,
+    )
+    blk.membrane_cost = pyo.Var(
+        initialize=30,
+        doc="Membrane cost",
+        units=pyo.units.USD_2018 / (pyo.units.meter**2),
+    )
+    blk.high_pressure_membrane_cost = pyo.Var(
+        initialize=50,
+        doc="Membrane cost",
+        units=pyo.units.USD_2018 / (pyo.units.meter**2),
+    )
```

Comment:
Do we have a source of these numbers, or are the just notional? If there is a source, we should include it.

Reply:
We are following previous paper of Tim and Adam, they provided values for the OARO modules. @adam-a-a can you indicate me which paper is better for reference?

Reply:
@bknueven , I added citations for the source of these costing parameter values